server/model/status_page.js

```diff
@@ -152,6 +173,105 @@ class StatusPage extends BeanModel {
         return $.root().html();
     }
 
+    /**
+     * render SVG file
+     * @param {slug} slug from router
+     * @param {userConfig} userConfig generated by user
+     * @param {statusPage} statusPage object
+     * @returns {Promise<string>} a SVG string
+     */
+    static async renderSVG(slug, userConfig = {}, statusPage) {
+        let allowedBadgeStyles = [ "flat", "flat-square", "for-the-badge", "plastic", "social" ];
+        if (!allowedBadgeStyles.includes(userConfig.badgeStyle)) {
+            userConfig.badgeStyle = "flat";
+        }
+
+        // returns unknown svg
+        if (statusPage === null) {
+            return makeBadge({
+                message: userConfig.notFoundText || "not found",
+                color: "red",
+                style: userConfig.badgeStyle,
+                label: slug,
+                labelColor: "gray"
+            });
+        } else {
+
+            // get all heartbeats that correspond to this statusPage
+            const config = await statusPage.toPublicJSON();
+
+            // Public Group List
+            const showTags = !!statusPage.show_tags;
+
+            const list = await R.find("group", " public = 1 AND status_page_id = ? ORDER BY weight ", [
+                statusPage.id
+            ]);
+
+            let heartbeats = [];
+
+            for (let groupBean of list) {
+                let monitorGroup = await groupBean.toPublicJSON(showTags, config?.showCertificateExpiry);
+                for (const monitor of monitorGroup.monitorList) {
+                    const heartbeat = await R.findOne("heartbeat", "monitor_id = ? ORDER BY time DESC", [ monitor.id ]);
+                    if (heartbeat) {
+                        heartbeats.push({
+                            ...monitor,
+                            status: heartbeat.status,
+                            time: heartbeat.time
+                        });
+                    }
+                }
+            }
+
+            // calculate RSS feed description
+            let status = StatusPage.overallStatus(heartbeats);
+
+            switch (status) {
```

Comment:
not a status page and not a description. lets inline

```suggestion
            switch (StatusPage.overallStatus(heartbeats)) {
```